src/mappings/entities/mapSettlement.ts

```diff
@@ -31,175 +38,212 @@ const finalizedEvents: EventIdEnum[] = [
   EventIdEnum.InstructionFailed,
 ];
 
-// Translates events into a settlements table. This includes transfers between a users portfolio combined with completed Instructions.
-export async function mapSettlement(
+const eventResultMap = {
+  [EventIdEnum.InstructionExecuted]: SettlementResultEnum.Executed,
+  [EventIdEnum.InstructionRejected]: SettlementResultEnum.Rejected,
+  [EventIdEnum.InstructionFailed]: SettlementResultEnum.Failed,
+  default: SettlementResultEnum.None,
+};
+
+export const createLeg = async (
   blockId: string,
-  eventId: EventIdEnum,
-  moduleId: ModuleIdEnum,
-  params: Codec[],
-  event: SubstrateEvent
-): Promise<void> {
-  if (moduleId === ModuleIdEnum.Portfolio && eventId === EventIdEnum.MovedBetweenPortfolios) {
-    await handlePortfolioMovement(blockId, eventId, params, event);
+  instructionId: string,
+  settlementId: string,
+  legIndex: number,
+  ticker: string,
+  amount: bigint,
+  from: Pick<Portfolio, 'identityId' | 'number'>,
+  to: Pick<Portfolio, 'identityId' | 'number'>
+): Promise<void> => {
+  await Promise.all([getPortfolio(from), getPortfolio(to)]);
+
+  return Leg.create({
+    id: `${blockId}/${instructionId ?? 'movePortfolio'}/${legIndex}`,
```

Comment:
I too don't like this but was not sure how to make it unique? Other option was to create an instruction in case of event - `MovedBetweenPortfolios`. 

What about directly - `${blockId}/${legIndex}` in this case?

Reply:
In that case i think `blockId/event.idx/legIndex` should work for sure